applications/GeoMechanicsApplication/custom_elements/small_strain_U_Pw_diff_order_element.cpp

```diff
@@ -173,6 +173,31 @@ Vector SmallStrainUPwDiffOrderElement::GetPressures(const size_t n_nodes) const
     return pressure;
 }
 
+void set_arithmetic_average_pressure(Geometry<Node>&                               rGeometry,
+                                     const Vector&                                 rPressure,
+                                     const std::vector<std::pair<size_t, size_t>>& index_pairs,
+                                     size_t                                        dest_offset = 0)
+{
+    for (size_t i = 0; i < index_pairs.size(); ++i) {
+        GeoElementUtilities::ThreadSafeNodeWrite(
+            rGeometry[dest_offset + i], WATER_PRESSURE,
+            0.5 * (rPressure[index_pairs[i].first] + rPressure[index_pairs[i].second]));
+    }
```

Comment:
I haven't tried it, but I think you could do something like this:

```suggestion
    for (size_t i= 0; const auto&[first_index, second_index] : index_pairs) {
        GeoElementUtilities::ThreadSafeNodeWrite(
            rGeometry[dest_offset + i], WATER_PRESSURE,
            0.5 * (rPressure[first_index] + rPressure[second_index]));
            ++i;
    }
```
But I leave it up to you if you think this is clearer

Reply:
Thanks a lot, this is clearer definitely.